### PR TITLE
Move students: hook up new component, remove old code

### PR DESCRIPTION
**Pull request by maddiedierker**
This PR is the last in a series to convert 'move students' from Angular to React. See my [previous PR](https://github.com/code-dot-org/code-dot-org/pull/22190) or the [spec](https://docs.google.com/document/d/18qEq8lQIvPrpMaWwCeAPsCXAPvYwVS7NYoMIt7_swvY/edit?usp=sharing) for context.

### What does this code do?
- updates/tweaks styling
- updates tests/storybook
- makes 'move students' table sort alphabetically by student name on dialog open (picture below)
- adds success notifications to parent component (more info + pictures below)
- removes old code
- updates `/sections/transfers` endpoint. specifically, the angular implementation was sending the `stay_enrolled_in_current_section` and `student_ids` params as stringified boolean and array of integers, respectively. the new implementation sends a JSON request with boolean/[int].

#### Manage Students tab
Routing to the 'manage students' tab, you should now see the new 'move students' button:
screen shot 2018-05-04 at 2 35 40 pm

#### Default alphabetical student sorting
On opening the dialog, the students in the dialog table should be sorted alphabetically by name:
screen shot 2018-05-04 at 2 05 37 pm

#### Success Notifications
When students are successfully transferred to a different section, the dialog should close and the `ManageStudentsTable` component (parent of `MoveStudents`) will render a notification. This notification should be different based on whether the students were *moved* (and subsequently removed from the current section) or *copied* to the new section.

If students were moved to new section:
screen shot 2018-05-04 at 2 18 23 pm

If students were copied to new section:
screen shot 2018-05-04 at 2 34 16 pm

### CODE REVIEW

File: apps/src/sites/code.org/pages/public/teacher-dashboard/index.js [hunk 1]
```diff
@@ -606,101 +602,6 @@ function main() {
...
-    };
-
-    $scope.getStayEnrolledInCurrentSection = function () {
-      if ($scope.selectedSectionCode == $scope.otherTeacher) {
-        return $scope.stayEnrolledInCurrentSection;
-      } else {
-        return false;
-      }
-    };
-  }]);
```
Comment (islemaster) [inline apps/src/sites/code.org/pages/public/teacher-dashboard/index.js:702]: threw_it_on_the_ground

File: apps/test/unit/templates/manageStudents/MoveStudentsTest.js [hunk 2]
```diff
@@ -62,19 +62,31 @@ describe('MoveStudents', () => {
...
+    const wrapper = mount(
+      <MoveStudents {...DEFAULT_PROPS}/>
+    );
+
+    wrapper.find('Button').simulate('click');
+    wrapper.find('#uitest-name-header').simulate('click');
+    const nameCells = wrapper.find('.uitest-name-cell');
+    expect(nameCells.at(0).text()).to.equal('studentc');
+    expect(nameCells.at(1).text()).to.equal('studentb');
+    expect(nameCells.at(2).text()).to.equal('studenta');
```
Comment (islemaster) [inline apps/test/unit/templates/manageStudents/MoveStudentsTest.js]: We're also using [`chai-enzyme`](https://github.com/producthunt/chai-enzyme#textstr) which enables enzyme-specific assertions like:

```js
expect(nameCells.at(0)).to.have.text('studentc');
expect(nameCells.at(1)).to.have.text('studentb');
expect(nameCells.at(2)).to.have.text('studenta');
```

**Comment by maddiedierker** [comment]
@islemaster thanks for the feedback! the `transfers_controller` tests should be good to go now 😃

File: apps/src/templates/manageStudents/ManageStudentsTable.jsx [hunk 3]
```diff
@@ -143,6 +160,40 @@ class ManageStudentsTable extends Component {
...
+  };
+
+  studentDataMinusBlanks = () => {
+    return this.props.studentData.filter(sd => sd.rowType === RowType.STUDENT);
+  };
+
+  isMoveStudentsEnabled = () => {
+    const {loginType} = this.props;
+    return loginType === SectionLoginType.word || loginType === SectionLoginType.picture || loginType === SectionLoginType.email;
+  };
```
Comment (islemaster) [inline apps/src/templates/manageStudents/ManageStudentsTable.jsx:195]: Sorry, one more thing: Can we add a test to [`ManageStudentsTableTest`](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js) asserting this behavior? (Or maybe its inverse - we don't show a move students button for Google or Clever sections.)

Comment (maddiedierker) [reply apps/src/templates/manageStudents/ManageStudentsTable.jsx:195]: for sure, that's a good call 👍

**Pull request merged**